cpp/ql/lib/semmle/code/cpp/commons/Printf.qll

```diff
@@ -359,6 +376,17 @@ private int lengthInBase10(float f) {
   result = f.log10().floor() + 1
 }
 
+private BufferWriteEstimationReason getEstimationReasonForIntegralExpression(Expr expr) {
```

Comment:
I'm not sure `expr` is sufficiently bound in this predicate, but let's test this with DCA before we make any changes. I'd be much less nervous about this if we added a `bindingset[expr]` to it.

Reply:
If this is a problem we might see it on performance? Should I add the tuple counting option?

Reply:
Yeah, you can try adding tuple counting. Be sure to update DCA to get the latest updates to that feature.